Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs

```diff
@@ -126,5 +127,231 @@ public void OpeningSvc_GetOpenings()
 
             }
         }
+
+        [TestMethod]
+        public void OpeningSvc_RequestOpening()
+        {
+            var s = new OpeningSvc();
+            using (var db = new AESDbContext())
+            {
+                var newStore = new Store()
+                {
+                    Name = "Test Store",
+                    Address = "Test Address",
+                    City = "Test City",
+                    Phone = "123-456-7654",
+                    State = "OR",
+                    Zip = 88888
+                };
+
+                db.Stores.RemoveRange(db.Stores.Where(st => st.Name == newStore.Name));
+                db.Stores.AddOrUpdate(newStore);
+                db.SaveChanges();
+
+                var newJob = new Job()
+                {
+                    Title = "Openings Test Job Title",
+                    ShortDescription = "Openings Test Job Short Description",
+                    LongDescription = "Openings Test Job Long Description"
+                };
+
+                db.Jobs.RemoveRange(db.Jobs.Where(j => j.Title == newJob.Title));
+                db.Jobs.AddOrUpdate(newJob);
+                db.SaveChanges();
+
+                string notes = "Test Request Notes";
+
+                var openingRequest = new JobOpeningContract()
+                {
+                    JobID = newJob.JobID,
+                    LongDescription = newJob.LongDescription,
+                    ShortDescription = newJob.ShortDescription,
+                    title = newJob.Title,
+                    RequestNotes = notes
+                };
+
+                Assert.IsTrue(s.RequestOpening(newStore.ID, openingRequest));
+                Assert.AreEqual(1, db.JobOpenings.Count(o => o.Status == OpeningStatus.PENDING_APPROVAL && o.Job.JobID == newJob.JobID));
+                Assert.IsTrue(s.RequestOpening(newStore.ID, openingRequest));
+                Assert.AreEqual(2, db.JobOpenings.Count(o => o.Status == OpeningStatus.PENDING_APPROVAL && o.Job.JobID == newJob.JobID));
+
+                var pendingOpenings = s.GetPendingOpenings(newStore.ID);
+                var allOpenings = s.GetAllOpenings(newStore.ID);
+                var approvedOpenings = s.GetApprovedOpenings(newStore.ID);
+                var rejectedOpenings = s.GetRejectedOpenings(newStore.ID);
+
+                Assert.AreEqual(2, pendingOpenings.Count());
+                Assert.AreEqual(2, allOpenings.Count());
+                Assert.AreEqual(0, approvedOpenings.Count());
+                Assert.AreEqual(0, rejectedOpenings.Count());
+
+                db.JobOpenings.RemoveRange(db.JobOpenings.Where(o => o.Job.JobID == newJob.JobID));
+
+                db.SaveChanges();
+
+                db.Jobs.RemoveRange(db.Jobs.Where(j => j.JobID == newJob.JobID));
+
+                db.SaveChanges();
+
+                db.Stores.RemoveRange(db.Stores.Where(st => st.ID == newStore.ID));
+
+                db.SaveChanges();
```

Comment:
I don't want to remove all openings, just the ones that match the filter I gave. Still, I removed the redundant saves in my next push.